packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -22,6 +30,14 @@ export default util.createRule<Options, 'conditional' | 'voidReturn'>({
     messages: {
       voidReturn:
         'Promise returned in function argument where a void return was expected.',
+      voidReturnVariable:
+        'Promise returned in variable where a void return was expected.',
```

Comment:
These new error messages aren't quite right, I think. The report is not that there's a Promise returned in the new cases -- it's that an async function is being given to a place that expects a void return.

Maybe:

```suggestion
        'Promise-returning function provided to variable where a void return was expected.',
```

...and similar for the others?